In [1]:
import pandas as pd
df= pd.read_csv("allbacks.csv")

In [2]:
df.shape,
df.head()

,rownames,volume,area,weight,cover
0,1,885,382,800,hb
1,2,1016,468,950,hb
2,3,1125,387,1050,hb
3,4,239,371,350,hb
4,5,701,371,750,hb


In [3]:
df.tail()

,rownames,volume,area,weight,cover
10,11,1492,0,975,pb
11,12,419,0,350,pb
12,13,1010,0,950,pb
13,14,595,0,425,pb
14,15,1034,0,725,pb


In [4]:
df.dtypes

rownames    int64
volume      int64
area        int64
weight      int64
cover         str
dtype: object

### Find the median of cover types

In [5]:
##sort cover column
sorted_cover = df["cover"].sort_values()
##grab the middle value
median_cover = sorted_cover.iloc[len(sorted_cover)//2]
print(f"The median cover is {median_cover}")


The median cover is pb


#### Verify Median Cover type by hand

In [6]:
print(df["cover"].value_counts())

cover
pb    8
hb    7
Name: count, dtype: int64


hb, hb, hb, hb, hb, hb, hb, pb, pb, pb, pb, pb, pb, pb, pb  
~~hb~~, ~~hb~~, ~~hb~~, ~~hb~~, ~~hb~~, ~~hb~~, ~~hb~~, pb, ~~pb~~, ~~pb~~, ~~pb~~, ~~pb~~, ~~pb~~, ~~pb~~, ~~pb~~ 

There are 15 books total, when you put put them in alphabetical order, and remove 7 books from each side, you are left with paperback, which would matches the median value we got earlier.

### Mean weight of all books

In [7]:
df["weight"].mean()

np.float64(706.6666666666666)

The mean weight of all books is 706.67.

#### Verify mean by hand

In [8]:
df["weight"].sum()

np.int64(10600)

In [9]:
10600/15

706.6666666666666

I got the sum of all the values in the weight column, which came out to 10,600. Then I divided by the total number of books, 15, and got the same value as using .mean().

### Mean weight for each cover

In [10]:
df.groupby("cover")["weight"].mean()

cover
hb    796.428571
pb    628.125000
Name: weight, dtype: float64

The mean weight of paperback books is 628.13, while the hardcover books mean weight is 796.43. This shows the type of cover greatly affects the weight of the book.

#### Verifying mean values by hand

In [11]:
df.groupby("cover")["weight"].sum()

cover
hb    5575
pb    5025
Name: weight, dtype: int64

In [12]:
hb_mean = 5575/7
pb_mean = 5025/8
print(f"The mean weight for hardcover books is {hb_mean}")
print(f"The mean weight for paperback books is {pb_mean}")


The mean weight for hardcover books is 796.4285714285714
The mean weight for paperback books is 628.125
